free-3.md

```diff
@@ -0,0 +1,230 @@
+# Free project 2++
+
+[I see **trees** of green](https://www.youtube.com/watch?v=A3yCcXgbKrE)
+
+## The user and a language
+This section describes who the project would serve and why a language might be a
+good way to meet their needs.
+
+
+### What's the need?
+
+The users I have in mind for this DSL are programmers who are interested in
+rapidly writing binary tree algorithms. They want to be able to test out ideas
+quickly and elegantly, and may not care very much about systems-level
+optimizations.
+
+Right now they people probably write a lot of boilerplate code for each tree
+they write, including the structure of nodes, code for tree rotations,
+debugging tools and more.
+
+Furthermore the details of some of the boilerplate may vary between trees
+written by a single author and definitely vary between trees written by
+different authors, making it just a bit harder to understand a tree written by
+someone else.
+
+I think that standardizing the way in which trees are written would help users
+understand what each other have written and would also save a lot of time spent
+writing boilerplate. This would also help users experiment with different ideas
+for tree algorithms more quickly.
```

Comment:
I'm interpreting this as suggesting that people use library implementations instead of writing their own. In many [basically all] applied situations that is the right call, but this tool is more aimed at people who are intentionally doing their own implementations, perhaps to explore new algorithm ideas.